src/expr.rs

```diff
@@ -1012,3 +1047,58 @@ fn rewrite_unary_op(context: &RewriteContext,
 
     Some(format!("{}{}", operator_str, subexpr))
 }
+
+fn rewrite_assign(context: &RewriteContext,
```

Comment:
nit: the names rewrite_assign and rewrite_assignment are too close and it is a bit confusing, perhaps chang rewrite_assignment to rewrite_assign_rhs or something more more descriptive?